nixos/modules/system/boot/loader/limine/limine.nix

```diff
@@ -0,0 +1,271 @@
+  ];
+
+  options.boot.loader.limine = {
+    enable = lib.mkEnableOption null // {
+      description = "Wheter to enable the Limine Bootloader.";
+    };
+
+    enableEditor = lib.mkEnableOption null // {
+      description = ''
+        Whether to allow editing the boot entries before booting them.
+        It is recommended to set this to false, as it allows gaining root
+        access by passing `init=/bin/sh` as a kernel parameter.
+      '';
+    };
+
+    maxGenerations = lib.mkOption {
+      default = null;
+      example = 50;
+      type = lib.types.nullOr lib.types.int;
+      description = ''
+        Maximum number of latest generations in the boot menu.
+        Useful to prevent boot partition of running out of disk space.
+        `null` means no limit i.e. all generations that were not
+        garbage collected yet.
+      '';
+    };
+
+    extraConfig = lib.mkOption {
+      default = "";
+      type = lib.types.lines;
+      example = lib.literalExpression ''
+        serial: yes
+      '';
+      description = ''
+        A string which is prepended to limine.conf. The config format can be found [here](https://github.com/limine-bootloader/limine/blob/trunk/CONFIG.md).
+      '';
+    };
+
+    extraEntries = lib.mkOption {
+      default = "";
+      type = lib.types.lines;
+      example = lib.literalExpression ''
+        /memtest86
+          protocol: chainload
+          path: boot():///efi/memtest86/memtest86.efi
+      '';
+      description = ''
+        A string which is appended to the end of limine.conf. The config format can be found [here](https://github.com/limine-bootloader/limine/blob/trunk/CONFIG.md).
+      '';
+    };
+
+    additionalFiles = lib.mkOption {
+      default = { };
+      type = lib.types.attrsOf lib.types.path;
+      example = lib.literalExpression ''
+        { "efi/memtest86/memtest86.efi" = "''${pkgs.memtest86-efi}/BOOTX64.efi"; }
+      '';
+      description = ''
+        A set of files to be copied to {file}`/boot`. Each attribute name denotes the
+        destination file name in {file}`/boot`, while the corresponding attribute value
+        specifies the source file.
+      '';
+    };
+
+    validateChecksums = lib.mkEnableOption null // {
+      default = true;
+      description = ''
+        Whether to validate file checksums before booting.
+      '';
+    };
+
+    panicOnChecksumMismatch = lib.mkEnableOption null // {
+      description = ''
+        Whether or not checksum validation failure should be a fatal
+        error at boot time.
+      '';
+    };
+
+    package = lib.mkPackageOption pkgs "limine" { };
+
+    efiSupport = lib.mkEnableOption null // {
+      default = pkgs.stdenv.hostPlatform.isEfi;
+      defaultText = lib.literalExpression "pkgs.stdenv.hostPlatform.isEfi";
+      description = ''
+        Whether or not to install the limine EFI files.
+      '';
+    };
+
+    efiInstallAsRemovable = lib.mkEnableOption null // {
+      default = true;
```

Comment:
Why is this true by default?

Reply:
Because before adding the option that was the default behavior.

Reply:
Ah sorry, I understood the option wrong.

Reply:
But you are right GRUB doesn't do this by default.

Reply:
Should this default to false?

Reply:
In my opinion, it should default to `efi.canTouchEfiVars` to prevent unbootable configurations. So systems where we can't touch EFI vars, we would install to EFI/BOOT so the firmware would still find it